SofaKernel/framework/sofa/simulation/BaseSimulationExporter.cpp

```diff
@@ -0,0 +1,169 @@
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+/******************************************************************************
+*       SOFA, Simulation Open-Framework Architecture, development version     *
+*                (c) 2006-2017 INRIA, USTL, UJF, CNRS, MGH                    *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+/******************************************************************************
+ *  Contributors:
+ *    - damien.marchal@univ-lille1.fr
+ *****************************************************************************/
+#include <sofa/simulation/BaseSimulationExporter.h>
+#include <sofa/simulation/AnimateBeginEvent.h>
+#include <sofa/simulation/AnimateEndEvent.h>
+#include <sofa/helper/system/FileSystem.h>
+
+namespace sofa
+{
+
+namespace simulation
+{
+
+namespace _basesimulationexporter_
```

Comment:
I see you often use underscores in namespaces, could you explain me why?

Reply:
Of course.

I'm using  dual namespace to separate what object are public and private in my .h/.cpp as this allow to employ the using keywords in .h without polluating the global namespaces. 

Eg: 
```cpp
namespace sofa  {
   namespace components {
       namespace myprivatens {
            using whatIWantWithoutPolluatingTheNS;
            class BlahBlah1{};
            class BlahBlah2{};
            class PrivateBlahBlah{};
     }
     /// Then "exxport" the public object
     using myprivatens::BlahBlah1;
     using myprivatens::BlahBlah2;
  }
} 

To make clear that the internal namespace is "private" I'm using a naming convention. 

The convention I know about this kind of thing is the qt one which look like: 
```cpp
namespace myfile
{
     namespace Private { ... };
}
```

Or the python way, that use underscore to indicate something is private. Looking more like 
```
namespace __myfile__ { ... };
```

I use arbitrarily to use the python convention.

Reply:
Very clear explanation, thanks! 👍